shared/src/main/scala/amf/core/model/domain/NamedDomainElement.scala

```diff
@@ -16,6 +17,14 @@ trait NamedDomainElement extends DomainElement {
   /** Return DomainElement name. */
   def name: StrField = fields.field(nameField)
 
+  def withName(node: YNode): this.type = withName(ScalarNodeObj(node))
+
+  def withName(name:String, a:Annotations): this.type = set(nameField, AmfScalar(name, a), Annotations.inferred())
+
   /** Update DomainElement name. */
-  def withName(name: String, a: Annotations = Annotations()): this.type = set(nameField, AmfScalar(name, a))
+  def withName(nameNode: ScalarNodeObj): this.type = set(nameField, nameNode.text(), Annotations.inferred())
+
+  def synthesizeName(name:String): this.type = set(nameField, AmfScalar(name), Annotations(SynthesizedField()))
```

Comment:
"withSyntheticName" is much more readable especially in the context of the "fold" where it is used.

Reply:
Agree